spark-commons/src/main/scala/za/co/absa/spark/commons/errorhandling/ErrorHandling.scala

```diff
@@ -16,25 +16,130 @@
 
 package za.co.absa.spark.commons.errorhandling
 
-import org.apache.spark.sql.{Column, DataFrame, SparkSession}
+import org.apache.spark.sql.catalyst.expressions.{CaseWhen, Expression}
+import org.apache.spark.sql.functions.when
+import org.apache.spark.sql.{Column, DataFrame}
 import za.co.absa.spark.commons.errorhandling.implementations.submits.{ErrorMessageSubmitOnColumn, ErrorMessageSubmitWithoutColumn}
 import za.co.absa.spark.commons.errorhandling.types._
 
+/**
+ * The basic class of error handling component. Every library that wants to use the component during Spark data
+ * processing should utilize this trait and its methods. The methods serve to record the errors and attach them to the
+ * `DataFrame`. The trait should be an input parameter for such library, perhaps as an implicit.
+ * On the other side the end application provides concrete `ErrorHandling` implementation, that does the actual error
+ * handling by the application desire.
+ * For easy to use ana as examples, a few general implementations are provided in the implementations sub-folder.
+ */
 trait ErrorHandling {
-  def putErrorToColumn(errorMessageSubmit: ErrorMessageSubmit): ErrorColumn
+  /**
+   * First of the few methods that needs to be coded in the trait implementation
+   * The purpose of this method is to convert the error specification into a `Column` expression
+   * @param errorMessageSubmit - the error specification
+   * @return - the error specification transformed into a column expression
+   * @group Error Handling
+   * @since 0.6.0
+   */
+  protected def transformErrorSubmitToColumn(errorMessageSubmit: ErrorMessageSubmit): Column
 
-  def aggregateErrorColumns(dataFrame: DataFrame)(errCols: ErrorColumn*): DataFrame
+  /**
+   * Applies the provided columns to the incoming DataFrame. Usually they might be aggregated in some way and attached
+   * to the DataFrame, but any other operations are imaginable. Unless really bent, the incoming columns are those
+   * produced by [[transformErrorSubmitToColumn]].
+   * The idea here is that the error column contains information of the error that occurred on the row or is empty (NULL)
+   * otherwise.
+   * In each implementation calling the function to each column separately or in any grouping of columns should produce
+   * the same result (with the exception of order of errors in the aggregation).
+   * @param dataFrame - the data frame to apply the error columns to
+   * @param errCols - the list of error columns to apply
+   * @return - data frame with the error columns applied (aggregated and attached or done otherwise)
+   */
+  protected def doApplyErrorColumnsToDataFrame(dataFrame: DataFrame, errCols: Column*): DataFrame
```

Comment:
`doApplyErrorColumnsToDataFrame` -> `applyErrorColumnsToDataFrame` can be good enough